packages/dds/tree/src/simple-tree/simpleSchema.ts

```diff
@@ -0,0 +1,136 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * Kind of {@link TreeNodeSchema}.
+ * @alpha
+ */
+export type SimpleNodeSchemaKind = "object" | "array" | "map" | "leaf";
+
+/**
+ * Kind of {@link FieldSchema}.
+ */
+export type SimpleFieldSchemaKind = "optional" | "required" | "identifier";
+
+/**
+ * Kind of leaf schema.
+ */
+export type SimpleLeafSchemaKind = "string" | "number" | "boolean" | "null" | "fluid-handle";
+
+/**
+ * Base interface for all {@link SimpleNodeSchema} implementations.
+ */
+export interface SimpleNodeSchemaBase<TNodeKind extends SimpleNodeSchemaKind> {
+	/**
+	 * The kind of {@link SimpleNodeSchema}.
+	 *
+	 * @remarks can be used to type-switch between implementations.
+	 */
+	readonly kind: TNodeKind;
+}
+
+/**
+ * A {@link SimpleNodeSchema} for an object node.
+ */
+export interface SimpleObjectNodeSchema extends SimpleNodeSchemaBase<"object"> {
+	/**
+	 * Schemas for each of the object's fields, keyed off of schema's view keys.
+	 */
+	readonly fields: Record<string, SimpleFieldSchema>;
```

Comment:
SimpleArrayNodeSchema.allowedTypes and SimpleMapNodeSchema.allowedTypes both store the identifiers for their types rather than the SimpleFieldSchema object itself. Should this record not do the same?

Reply:
Object schemas are special because they have "fields" (that can be required/optional, etc.). This preserves that specialness in the "simple schema" domain. The field schema then has the required/optional details, and the referenced identifier.